semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -141,7 +141,14 @@ trait TextDocumentOps {
             names.result()
           }
           private def indexArgNames(mapp: m.Tree): Unit = mapp match {
-            case t: m.Init => margnames(t.pos.start) = getAssignLhsAndIndexRhs(t.argClauses)
+            case t: m.Init => 
+              // 2.13.16 changed position
+              t.parent.foreach{
+                case n: m.Term.New =>
+                  margnames(n.pos.start) = getAssignLhsAndIndexRhs(t.argClauses)
+              }
+              margnames(t.pos.start) = getAssignLhsAndIndexRhs(t.argClauses)
```

Comment:
i am not sure if `foreach` will not throw if parent is not `new`... also, we call `getAssign...` twice.

perhaps this is safer:
```
val tree = t.parent match {
  case Some(p: m.Term.New) => p
  case _ => t
}
margnames(tree.pos.start) = getAssignLhsAndIndexRhs(t.argClauses)
```
unless you want to set to both, in which case:
```
val names = getAssignLhsAndIndexRhs(t.argClauses)
margnames(t.pos.start) = names
t.parent match {
  case Some(p: m.Term.New) => margnames(p.pos.start) = names
  case _ =>
}
```

Reply:
should we also do it for `Ctor.Block` and `Mod.Annot`? the code that uses this is:
```
            if (gsym != null && (gsym.isMethod || gsym.isConstructor)) for {
              margnames <- margnames.get(gstart) ++ margnames.get(gpoint)
              margname <- margnames
              gparams <- gsym.paramss
              gparam <- gparams.find(_.name.decoded == margname.value)
            } success(margname, gparam)
```
so it's any method or ctor with argument names.

p.s. perhaps there's a simpler way to do this...

Reply:
Looks like the issue only happens with New, but I can check other possible problems

Reply:
I added some tests to TargetedSuite and it seems to work. I can't figure out how to get Ctor.Body actually

Reply:
Looks like it's already covered in the test actually. I was using an older version of scalameta